### ISSUE TITLE
magnet link hangs

### ISSUE THREAD

**Issue by LLloody** [SuRGeoNix/BitSwarm#14]
First: keep up the good work!
.open() does nothing with the following magnet link even in the WinForms demo.
magnet:?xt=urn:btih:ee17eb980ac506e3043474bf3fb42a7777ad991b&dn=Bitchin.Rides.S09E08.Is.There.Anything.Good.on.HDTV.x264-CRiMSON%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2760&tr=udp%3A%2F%2F9.rarbg.to%3A2780&tr=udp%3A%2F%2Ftracker.fatkhoala.org%3A13750&tr=udp%3A%2F%2Ftracker.thinelephant.org%3A12740

no error or other indication MetadataReceived never fires (or ANY of the events)

**Comment by SuRGeoNix** [comment]
just tested winforms demo and works fine

**Comment by LLloody** [comment]
I have since discovered..
.open() will trigger MetadataReceived event IF opening a local .torrent on the machine.
I was using that event to prompt the user for which files to include THEN calling .Start()
This strategy won't work with a magnet link.
On another note: does .open() work with a http .torrent link or do I need to download locally first?

**Comment by SuRGeoNix** [comment]
Open will Open a Torrent File, Magnet Link, SHA-1 Hex Hash & Base32 Hash. It will not open an http torrent you will need to download it first.
winforms demo does exactly that so I'm not sure what your issue is. Use the debugger and let me know if you need more help.

**Comment by LLloody** [comment]
No problem bud! I had modified the demo and moved .Start() to the MetadataReceived event.
I tested with a local Torrent File and it still worked. I had not tested with a magnet or hash.
The reason I had changed it was because some torrents have small .nfo and other garbage files.
I wanted to avoid those being downloaded BEFORE the user had a chance to exclude them.
If you feel like making a change so that .Open() will parse and get metadata for magnets and hash
the way it does for Torrent File, that would be cool.
Either way I will close this issue.
Thanks
Woody
